### ISSUE TITLE
[java] False-positive MisplacedNullCheck reported

### ISSUE THREAD

**Issue by pzygielo** [pmd/pmd#2242]
**Affects PMD Version:** 6.20.0

**Rule:** [MisplacedNullCheck](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#misplacednullcheck)

**Description:**
Violation reported for the line 20 (in method `triggerPar`) of example below.
For the same line [UselessParentheses](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#uselessparentheses) is also reported (properly). When that violation is cleared, as done in example in next method (`triggerNoPar`) - _MisplacedNullCheck_ is no longer reported there, for condition is exactly the same.

**Code Sample demonstrating the issue:**
[Issue.java#L20](https://github.com/pzrep/pmd-MisplacedNullCheck/blob/29abcbf49c6d5d1d567e0de8007d8f1da7fd6fc7/src/main/java/pzrep/Issue.java#L20):

```java
     public boolean triggerPar(String l) {
        boolean c = l != null
20:             && ((h != null && !l.equals(h.getName())) || h == null);
        return c;
     }

     public boolean triggerNoPar(String l) {
        boolean c = l != null
                && (h != null && !l.equals(h.getName()) || h == null);
        return c;
     }
```

**Running PMD through:** *CLI*, *Maven*
[CLI `-f txt` output](https://github.com/pzrep/pmd-MisplacedNullCheck/commit/29abcbf49c6d5d1d567e0de8007d8f1da7fd6fc7/checks?check_suite_id=410456577#step:8:9)

In xml output, all 3 (different question - why 3?) violations refer to the same line and column
```xml
<violation beginline="20" endline="20" begincolumn="42" endcolumn="50" rule="MisplacedNullCheck" ruleset="Error Prone" package="pzrep" class="Issue" method="triggerPar" externalInfoUrl="https://pmd.github.io/pmd-6.20.0/pmd_rules_java_errorprone.html#misplacednullcheck" priority="3">
The null check here is misplaced; if the variable is null there will be a NullPointerException
</violation>
```

**Comment by adangel** [comment]
> different question - why 3?

This seems to be a bug in Jaxen. If we switch to XPath 2.0/Saxon, the found nodes are correctly merged.

**Comment by pzygielo** [comment]
@adangel thank you for investigation and the incoming fix!
